File: sample/src/androidTest/java/com/schibsted/spain/barista/sample/CheckBoxTest.java
package com.schibsted.spain.barista.sample;

import android.support.test.rule.ActivityTestRule;
import android.support.test.runner.AndroidJUnit4;
import com.schibsted.spain.barista.sample.util.FailureHandlerValidatorRule;
import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import static com.schibsted.spain.barista.BaristaAssertions.assertDisplayed;
import static com.schibsted.spain.barista.BaristaCheckBoxActions.clickCheckBoxItem;

@RunWith(AndroidJUnit4.class)
public class CheckBoxTest {

  @Rule
  public ActivityTestRule<CheckBoxActivity> activityRule = new ActivityTestRule<>(CheckBoxActivity.class);

  @Rule
  public FailureHandlerValidatorRule handlerValidator = new FailureHandlerValidatorRule();

  @Test
  public void checkCheckBoxClickById_firstItem() {
    clickCheckBoxItem(R.id.first_item);
    assertDisplayed("" + R.id.first_item);
  }

  @Test
  public void checkCheckBoxClickById_secondItem() {
    clickCheckBoxItem(R.id.second_item);
    assertDisplayed("" + R.id.second_item);
  }

  @Test
  public void checkCheckBoxClickByText_firstItem() {
    clickCheckBoxItem("Hello");
    assertDisplayed("" + R.id.first_item);
  }

  @Test
  public void checkCheckBoxClickByText_secondItem() {
    clickCheckBoxItem("Bye bye");
    assertDisplayed("" + R.id.second_item);
  }
}

File: sample/src/androidTest/java/com/adevinta/android/barista/sample/CheckboxTest.kt
package com.adevinta.android.barista.sample

import androidx.test.rule.ActivityTestRule
import com.adevinta.android.barista.assertion.BaristaCheckedAssertions.assertChecked
import com.adevinta.android.barista.assertion.BaristaCheckedAssertions.assertUnchecked
import com.adevinta.android.barista.interaction.BaristaCheckboxInteractions.setChecked
import com.adevinta.android.barista.sample.util.SpyFailureHandlerRule
import org.junit.Rule
import org.junit.Test

class CheckboxTest {

  @get:Rule
  var activityRule = ActivityTestRule(CheckBoxActivity::class.java)

  @get:Rule
  var spyFailureHandlerRule = SpyFailureHandlerRule()

  @Test
  fun check_checkBox() {
    setChecked(R.id.first_item, true)

    assertChecked(R.id.first_item)
    assertUnchecked(R.id.second_item)
    spyFailureHandlerRule.assertNoEspressoFailures()
  }

  @Test
  fun uncheck_checkBox() {
    setChecked(R.id.first_item, true)
    setChecked(R.id.second_item, true)

    setChecked(R.id.first_item, false)

    assertUnchecked(R.id.first_item)
    assertChecked(R.id.second_item)
    spyFailureHandlerRule.assertNoEspressoFailures()
  }
}